contracts/HomoraIBSwap.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity ^0.7.0;
+
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/token/ERC20/IERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/token/ERC20/SafeERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/math/SafeMath.sol";
+import "../interfaces/IUniswapV2Router02.sol";
+import "../interfaces/ISafeBox.sol";
+import "../interfaces/ISafeBoxETH.sol";
+import "../interfaces/ICErc20.sol";
+import "../interfaces/ICyToken.sol";
+
+contract HomoraIBSwap {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    address private constant IBETHV2 =
+        0xeEa3311250FE4c3268F8E684f7C87A82fF183Ec1;
+    address private constant IBUSDTV2 =
+        0x020eDC614187F9937A1EfEeE007656C6356Fb13A;
+    address private constant IBUSDCV2 =
+        0x08bd64BFC832F1C2B3e07e634934453bA7Fa2db2;
+    address private constant IBDAIV2 =
+        0xee8389d235E092b2945fE363e97CDBeD121A0439;
+
+    address internal constant UNISWAP_ROUTER_ADDRESS =
+        0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    IUniswapV2Router02 public uniswapRouter;
+
+    constructor() {
+        uniswapRouter = IUniswapV2Router02(UNISWAP_ROUTER_ADDRESS);
+    }
+
+    function supported(address token) public returns (bool) {
+        if (
+            token == IBETHV2 ||
+            token == IBUSDTV2 ||
+            token == IBUSDCV2 ||
+            token == IBDAIV2
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    function path(uint256 order, address token)
```

Comment:
I think it's less confusing to make this function takes two addresses (one will be WETH, the other will be token) instead of `order` and `token`